boards/raspberry_pi_pico/src/main.rs

```diff
@@ -451,6 +459,42 @@ pub unsafe fn main() {
         rp2040::adc::Adc
     ));
 
+    // RTC DATE TIME
+
+    match peripherals.rtc.rtc_init() {
+        Ok(_) => {}
+        Err(e) => {
+            debug!("error starting rtc {:?}", e);
+        }
+    };
+
+    let date_time = components::date_time::DateTimeComponent::new(
+        board_kernel,
+        capsules_extra::date_time::DRIVER_NUM,
+        &peripherals.rtc,
+    )
+    .finalize(date_time_component_static!(rp2040::rtc::Rtc<'static>));
+
+    // uncomment for setting up current date and time
+    // let dt = DateTime {
+    //     year: 2021,
+    //     month: Month::January,
+    //     day: 11,
+    //     day_of_week: DayOfWeek::Monday,
+    //     hour: 13,
+    //     minute: 33,
+    //     seconds: 1,
+    // };
+    //
+    // match peripherals.rtc.set_date_time(dt) {
+    //     Err(e) => {
+    //         debug!("Error setting date and time {:?}", e);
+    //     }
+    //     _ => {
+    //         debug!("Date and time are set");
+    //     }
+    // };
```

Comment:
Leaving this here makes me nervous. Since the date time driver is already setup, that driver is going to get the callback when this operation finishes. Now, that is probably benign in practice, but, it's not good practice for code to get callbacks that it didn't request.

I don't think we should include this example in default tock boards because it isn't something we would like replicated.

Now, if there was a virtual_date_time.rs layer, then the kernel could do this safely.